### PR TITLE
Initial Kotlin support

### PR DESCRIPTION
**Pull request by igfoo**

### CODE REVIEW

**Review by michaelnebel — APPROVED** [review]
C#: LGTM

**Comment by yo-h** [comment]
> Conflicting files
> java/ql/lib/semmle/code/java/Expr.qll

**Review by atorralba — APPROVED** [review]

File: java/ql/lib/semmle/code/java/Expr.qll [hunk 1]
```diff
@@ -949,20 +972,54 @@ class GEExpr extends BinaryExpr, @geexpr {
...
+ */
+class ValueOrReferenceEqualsExpr extends BinaryExpr {
+  ValueOrReferenceEqualsExpr() { this instanceof EQExpr or this instanceof ValueEQExpr }
+}
+
+/**
+ * A binary expression using either Java or Kotlin's `!=` operator.
+ *
+ * This might test for reference equality or might function like `Objects.equals`. If you
+ * need to distinguish them, use `EQExpr` or `ValueEQExpr` instead.
```
Comment (intrigus-lgtm) [inline java/ql/lib/semmle/code/java/Expr.qll:1017]: This should be
> `NEExpr` or `ValueNEExp`

right?

File: java/ql/lib/semmle/code/java/Expr.qll [hunk 2]
```diff
@@ -2177,3 +2405,111 @@ class ValueDiscardingExpr extends Expr {
...
+  /** Gets the expression whose class is being returned. */
+  Expr getExpr() { result.isNthChildOf(this, 0) }
+
+  override string toString() { result = "::class" }
+
+  override string getAPrimaryQlClass() { result = "ClassExpr" }
+}
+
+/**
+ * An statement expression.
```
Comment (intrigus-lgtm) [inline java/ql/lib/semmle/code/java/Expr.qll:2464]: **A** statement.

Comment (igfoo) [reply java/ql/lib/semmle/code/java/Expr.qll:2464]: Thanks! I've made a PR for these here: https://github.com/github/codeql/pull/9136

**Pull request merged**